### ISSUE TITLE
Get index of rows in dataframe

### ISSUE THREAD

**Issue by ynjacobs** [haifengl/smile#603]
Thanks so much for this wonderful library!

I'm wondering if there is a way to get the indices of the rows of a dataframe in an array?
Similar to python pandas.DataFrame.index.

For ex. If the original dataframe is 
```
Status   Name       Age
 g        Tim        34
 g        Eric       70
 e        Nancy      45
 g        Lauren     33
 e        Peter      56
 e        Max        29
```
and I filter the dataframe to get only rows containing "g" `DataFrame onlyG = DataFrame.of(dfAllStatus.stream().filter(row -> row.getString("Status").equals("g")));`

is there a way to get an array of indices in `onlyG` [0, 1, 3] corresponding to the position of rows containing "g" in the original dataframe?

Thanks so much!

**Comment by haifengl** [comment]
no direct way. you may add an index column though.

**Comment by ynjacobs** [comment]
Thank you!

Is there a method that can add a new column name with rows being numbers 0 - dataframe.size() -1, without the need to modify the csv directly?

**Comment by haifengl** [comment]
`DataFrame.merge()`

**Comment by ynjacobs** [comment]
Great, thanks!

**Comment by ynjacobs** [comment]
As a follow up, is there a way to get a new dataframe by selecting a column and looping through an array to get the rows?

For ex. continuing from above, I added a new column index with rows of numbers. 

I'd like to get a new dataframe where rows in the index column = 0, 2, 3, 4, 6
`List <Integer> indexList = [0, 2, 3, 4, 6]`

I tried:
```
DataFrame dfStatusSelectIndex = null;
  for (Integer element : indexList) {
	dfStatusSelectIndex = DataFrame.of(dfAllStatus.stream()
	        	    .filter(row -> row.get(0).equals(element)));
	System.out.println(dfStatusSelectIndex);
  }
   System.out.println(dfStatusSelectIndex);
```

But when I print inside the loop I get back 5 separate 1 row dataframes for each index of element and when I print outside the loop I only get 1 row dataframe of the last element but I'd like 1 dataframe with all 5 rows.
